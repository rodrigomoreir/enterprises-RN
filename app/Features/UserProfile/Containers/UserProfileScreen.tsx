import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { View, Text, Image, ScrollView } from 'react-native';
import { getStyles } from './UserProfileScreen.style'
import { Button, Header } from '../../../Components'
import { useAuth } from '../../../Hooks/Auth'
import { Separator } from '../Components';
import images from '../../../Themes/Images';

const UserProfile: React.FC = () => {
  const navigation = useNavigation()
  const styles = getStyles()
  const navigationGoBack = () => navigation.goBack()

  const { signOut } = useAuth()

  return (
    <>
      <Header title={'Usuário'} goProfile={() => {}} goBack={navigationGoBack} />
      <ScrollView
        keyboardShouldPersistTaps='handled'
        style={styles.containerScrollView}
        contentContainerStyle={styles.contentScrollView}
      >   
        <View style={styles.container}>
          <View style={styles.imageContainer}>  
            <Image source={images.profileUser} style={styles.image} />
          </View>
          <Text style={styles.itemText}>Minhas Informações</Text>
          <Separator style={styles.separator}/>
          <Text style={styles.itemText}>Editar Cadastro</Text>
          <Separator style={styles.separator}/>
          <Text style={styles.itemText}>Empresa</Text>
          <Separator style={styles.separator}/>
          <Text style={styles.itemText}>Saldo</Text>
          <Separator style={styles.separator}/>
          <Text style={styles.itemText}>Valor do Portifolio</Text>
          <Separator style={styles.separator}/>
        </View>
        <View style={styles.buttonContainer}>
          <Button text={'Sair'} onPress={signOut} />
        </View>
      </ScrollView>
    </>
  )
}

export default UserProfile;
