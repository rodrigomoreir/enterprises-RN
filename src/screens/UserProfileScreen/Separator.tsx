import React from 'react'
import { StyleProp, ViewStyle } from 'react-native'
import styled from 'styled-components/native'
import themeColors from '../../theme/theme'
import addOpacity from '../../utils/addOpacity'

export const DEFAULT_COLOR_OPACITY = 0.5

interface Props {
  color?: string
  style?: StyleProp<ViewStyle>
}

const Separator = ({ color, style }: Props) => {
  return (
    <StyledSeparator
      accessibilityLabel={'Separator'}
      style={[style, !!color && { backgroundColor: addOpacity(color, 0.5) }]}
    />
  )
}

const StyledSeparator = styled.View`
  height: 1px;
  width: 100%;
  background-color: ${addOpacity(
  themeColors.colors.darkGrey,
  DEFAULT_COLOR_OPACITY,
)};
`

export default Separator
