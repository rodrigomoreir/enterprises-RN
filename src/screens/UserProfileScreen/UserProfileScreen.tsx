import React from 'react'
import { useNavigation } from '@react-navigation/native'
import { Image } from 'react-native'

import styled from 'styled-components/native'
import { Button, Header } from '../../core'
import Separator from './Separator'
import images from '../../assets'
import { screenWidth } from '../../utils'
import { useDispatch } from 'react-redux'
import { userActions } from '../../store/slices/userSlice'

const UserProfile: React.FC = () => {
  const navigation = useNavigation()
  const dispatch = useDispatch()
  const navigationGoBack = () => navigation.goBack()

  const signOut = () => {
    dispatch(userActions.logout())
  }

  return (
    <StyledSafeAreaView>
      <Header
        title={'Usuário'}
        goProfile={() => { }}
        goBack={navigationGoBack}
      />
      <StyledScrollView keyboardShouldPersistTaps="handled" bounces={false}>
        <StyledViewContainer>
          <StyledViewImage>
            <StyledImage source={images.profileUser} />
          </StyledViewImage>
          <StyledTextItem>Minhas Informações</StyledTextItem>
          <StyledSeparator />
          <StyledTextItem>Editar Cadastro</StyledTextItem>
          <StyledSeparator />
          <StyledTextItem>Empresa</StyledTextItem>
          <StyledSeparator />
          <StyledTextItem>Saldo</StyledTextItem>
          <StyledSeparator />
          <StyledTextItem>Valor do Portifolio</StyledTextItem>
          <StyledSeparator />
        </StyledViewContainer>
        <StyledViewButton>
          <Button text={'Sair'} onPress={() => signOut()} />
        </StyledViewButton>
      </StyledScrollView>
    </StyledSafeAreaView>
  )
}

const StyledSafeAreaView = styled.SafeAreaView`
  flex: 1;
`
const StyledScrollView = styled.ScrollView.attrs({
  contentContainerStyle: {
    flexGrow: 1,
    width: screenWidth,
    alignItems: 'center',
    justifyContent: 'space-between',
  },
})`
  width: ${screenWidth}px;
`
const StyledViewContainer = styled.View`
  flex: 1;
  align-items: center;
  justify-content: center;
`
const StyledViewImage = styled.View`
  width: 100px;
  height: 100px;
  align-items: center;
  justify-content: center;
  margin-bottom: 30px;
`
const StyledImage = styled(Image)`
  width: 100%;
  height: 100%;
  tint-color: ${({ theme }) => theme.colors.blue};
`
const StyledTextItem = styled.Text`
  align-self: center;
  margin-vertical: 10px;
  font-weight: bold;
  font-size: 20px;
  line-height: 26.38px;
  font-style: normal;
  color: ${({ theme }) => theme.colors.darkGrey};
  text-align: center;
`
const StyledSeparator = styled(Separator)`
  width: ${screenWidth - 50}px;
`
const StyledViewButton = styled.View`
  align-items: center;
  justify-content: center;
  margin-bottom: 20px;
`

export default UserProfile
