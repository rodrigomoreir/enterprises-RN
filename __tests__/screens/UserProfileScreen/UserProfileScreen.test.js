import React from 'react'
import UserProfileScreen from '../../../src/screens/UserProfileScreen'

import {
  renderWithTheme,
  matchSnapshotWithTheme,
} from '../../../src/functions/testing'

describe('User Profile Screen', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(<UserProfileScreen />)
  })

  test('renders correctly', () => {
    renderWithTheme(<UserProfileScreen />)
  })
})
