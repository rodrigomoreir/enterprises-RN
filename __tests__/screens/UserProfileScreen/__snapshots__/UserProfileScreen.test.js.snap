// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Profile Screen matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomWidth": 2,
          "borderColor": "#a9a9a9",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "paddingHorizontal": 10,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": 40,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#87cefaCC",
            "borderColor": "#69696999",
            "borderRadius": 20,
            "borderWidth": 1,
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "width": 40,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/chevronLeft.png",
            }
          }
          style={
            Array [
              Object {
                "height": 20,
                "tintColor": "#ecf0f2",
                "width": 20,
              },
            ]
          }
        />
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#696969",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": 26.38,
            "textAlign": "center",
          },
        ]
      }
    >
      Usuário
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#69696999",
          "borderRadius": 20,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/profile.png",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 20,
              "maxHeight": 38,
              "maxWidth": 38,
              "tintColor": "#87cefaCC",
            },
          ]
        }
      />
    </View>
  </View>
  <RCTScrollView
    bounces={false}
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "space-between",
        "width": 750,
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Array [
        Object {
          "width": 750,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 100,
                "justifyContent": "center",
                "marginBottom": 30,
                "width": 100,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/profile.png",
              }
            }
            style={
              Array [
                Object {
                  "height": "100%",
                  "tintColor": "#87cefa",
                  "width": "100%",
                },
              ]
            }
          />
        </View>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#696969",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "lineHeight": 26.38,
                "marginVertical": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Minhas Informações
        </Text>
        <View
          accessibilityLabel="Separator"
          style={
            Array [
              Object {
                "backgroundColor": "#69696980",
                "height": 1,
                "width": "100%",
              },
              Array [
                Object {
                  "width": 700,
                },
              ],
              false,
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#696969",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "lineHeight": 26.38,
                "marginVertical": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Editar Cadastro
        </Text>
        <View
          accessibilityLabel="Separator"
          style={
            Array [
              Object {
                "backgroundColor": "#69696980",
                "height": 1,
                "width": "100%",
              },
              Array [
                Object {
                  "width": 700,
                },
              ],
              false,
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#696969",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "lineHeight": 26.38,
                "marginVertical": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Empresa
        </Text>
        <View
          accessibilityLabel="Separator"
          style={
            Array [
              Object {
                "backgroundColor": "#69696980",
                "height": 1,
                "width": "100%",
              },
              Array [
                Object {
                  "width": 700,
                },
              ],
              false,
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#696969",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "lineHeight": 26.38,
                "marginVertical": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Saldo
        </Text>
        <View
          accessibilityLabel="Separator"
          style={
            Array [
              Object {
                "backgroundColor": "#69696980",
                "height": 1,
                "width": "100%",
              },
              Array [
                Object {
                  "width": 700,
                },
              ],
              false,
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#696969",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "lineHeight": 26.38,
                "marginVertical": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Valor do Portifolio
        </Text>
        <View
          accessibilityLabel="Separator"
          style={
            Array [
              Object {
                "backgroundColor": "#69696980",
                "height": 1,
                "width": "100%",
              },
              Array [
                Object {
                  "width": 700,
                },
              ],
              false,
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginBottom": 20,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#87cefa",
              "borderRadius": 30,
              "height": 60,
              "justifyContent": "center",
              "opacity": 1,
              "width": 650,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 18,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Sair
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
